### ISSUE TITLE
1.12.2: bugs with groups in .rs files

### ISSUE THREAD

**Issue by CryLegende** [kangarko/ChatControl#573]
Hello, groups in all .rs files didn't seem to work fine.
For example, if I put two groups, vip and staff, everyone will be in the vip group even if I negate the permission chatcontrol.group.vip to the players

Here everyone has the message from group vip, even the players with the permission chatcontrol.group.vip negated

```
group vip
message:
- &8&l» {player_prefix}{player} &6join the server !

group staff
message:
- &8&l» {player_prefix}{player} &6join the server ! &7(&3staff&7)
```

Minecraft Version: PaperSpigot 1.12.2
MySQL: false
BungeeCord: false

Thank you for your help :)

**Comment by ElBananaa** [comment]
Hey!
Please provide us your `/chc debug` output file.

**Comment by CryLegende** [comment]
https://ufile.io/duprg3vp :)

**Comment by ElBananaa** [comment]
> https://ufile.io/duprg3vp :)

Alright thanks!
So, from what I can see, you're not using correctly the conditions in the chat format file.
the `Sender_Condition` need to return either true or false.

Basically, instead of ```
  message_vip:
    Message: '&f{message} '
    Sender_Condition: '''{player_has_permission_chatcontrol.staff}'' == ''no'' && ''{player_has_permission_chatcontrol.vip}'' == ''yes'''
    Suggest_Command: '/mp {player_name} '```
for the vip rank, use:
```
  message_vip:
    Message: '&f{message} '
    Sender_Permission: chatcontrol.vip
    Suggest_Command: '/mp {player_name} '
```
for the staff rank use:
```
  message_staff:
    Message: '{vault_prefix_color}{message} '
    Sender_Permission: chatcontrol.staff
    Suggest_Command: '/mp {player_name} '
```
and for the default rank use:
```
  message:
    Message: '&7{message} '
    Sender_Permission: chatcontrol.player
    Suggest_Command: '/mp {player_name} '
```
Once this is done, deny the `chatcontrol.player` and `chatcontrol.vip` permissions for staff ranks, deny `chatcontrol.player` & `chatcontrol.staff` for the vip rank, and deny `chatcontrol.vip` and `chatcontrol.staff` for players.
(And ofc, don't forget to give their rank the right permission.

**Comment by ElBananaa** [comment]
Sorry, I was lost in my mind, just noticed I didn't help at all xD.
Regarding the join issue, simply use `require sender perm`.
You can find all the arguments that you can use in messages on the wiki page:
https://github.com/kangarko/ChatControl-Red/wiki/messages
```
group vip
require sender perm chatcontrol.group.vip
message:
- &8&l» {player_prefix}{player} &6join the server !

group staff
require sender perm chatcontrol.group.staff
message:
- &8&l» {player_prefix}{player} &6join the server ! &7(&3staff&7)
```

**Comment by CryLegende** [comment]
Thank you! I thought that the group was the one displayed in the plugin config but in fact it is useless in the rs files except to make categories ?

I will follow your instructions for the conditions ^^
> Sorry, I was lost in my mind, just noticed I didn't help at all xD.

**Comment by ElBananaa** [comment]
> Thank you! I thought that the group was the one displayed in the plugin config but in fact it is useless in the rs files except to make categories ?
> 
> I will follow your instructions for the conditions ^^
> 
> > Sorry, I was lost in my mind, just noticed I didn't help at all xD.

Yes, groups are useful to create a category of messages. And groups used in messages are different than the ones used in `rules/groups.rs`.

Anyways, I'm closing this issue, feel free to keep commenting down below if you still need help related to this, or open a new ticket if you need anything else.
